sdk/logical/event.proto

```diff
@@ -0,0 +1,38 @@
+syntax = "proto3";
+
+option go_package = "github.com/hashicorp/vault/sdk/logical";
+
+package logical;
+
+// EventPluginInfo contains data related to the plugin that generated an event.
+message EventPluginInfo {
+    // The type of plugin this event originated from, i.e., "auth" or "secrets.
+    string mount_class = 1;
+    // Unique ID of the mount entry, e.g., "kv_957bb7d8"
+    string mount_accessor = 2;
+    // Mount path of the plugin this event originated from, e.g., "secret/"
+    string mount_path = 3;
+    // Plugin name that this event originated from, e.g., "kv"
+    string plugin = 4;
+    // Plugin version of the plugin this event originated from, e.g., "v0.13.3+builtin"
+    string plugin_version = 5;
+    // Mount version that this event originated from, i.e., if KVv2, then "2". Usually empty.
+    string version = 6;
+}
+
+// EventData contains event data in a CloudEvents container.
+message EventData {
+    // EID identifies the event, cannot be an empty and is required. The
+    // combination of Source + EID must be unique. Events with the same Source +
+    // ID can be assumed to be duplicates by consumers.
+    // (Named EID to avoid conflict with ID method.)
+    string eid = 1;
+    // If the event if plugin-generated, this contains information on the plugin.
+    EventPluginInfo plugin_info = 2;
```

Comment:
I like having the plugin data all together, but this is kind of inverted from what I meant. Specifically in the plugin interface, we want plugins to be able to send Vault core events and set fields such as metadata, entity_ids, and note (and I think eid too IIUC), but we _don't_ want the plugin to populate all the fields in these structs, because the Vault core is the system of record for that info.

I'm not totally averse to just telling plugins not to set that because it will be overwritten, but it would feel cleaner if the field wasn't exposed to plugins at all (i.e. have another struct encapsulate what is currently called `EventData`, and make `EventData` and `EventPluginInfo` siblings in that struct - then the RPC call only includes `EventData` in the arguments). Again though, I haven't followed this logic all the way through to the impact on the API, so there might be other clunky bits it introduces.

Reply:
I think that makes sense. I'll try splitting this out as you suggested. Nothing is set in stone (thanks to experiments :) ) so we can always go back if it turns out to be too awkward. But there's not a big penalty for adding a wrapping protobuf in terms of encoding.